CRM/Contribute/Page/Tab.php

```diff
@@ -251,13 +260,6 @@ private function buildRecurringContributionsArray($recurContributions) {
       }
 
       if ($recurContributions[$recurId]['is_active']) {
-        $details = CRM_Contribute_BAO_ContributionRecur::getSubscriptionDetails($recurContributions[$recurId]['id'], 'recur');
-        $hideUpdate = $details->membership_id & $details->auto_renew;
```

Comment:
@mattwire this PR makes sense to me with an exception about the doubts of this one line.  It seems the intention of this line was to suppress the editing in the case where auto_renew is set (although the use of a bitwise operator is confusing!). 

This is described as.

'Some systems allow contributor to set a number of installments - but then auto-renew the subscription or commitment if they do not cancel.'

TBH I can barely understand what that means & the code rarely helps! However it seems the issue is not that it is being suppressed for memberships per se but that it is being suppressed for auto_renew memberships. So we need to try to figure out what the impact of auto_renew is....

Reply:
Right, this is kind of a "change" but it was already agreed and implemented in https://issues.civicrm.org/jira/browse/CRM-21512 ... It's just we missed this bit.  You have been able to navigate directly to the "Edit recurring contribution" page since CRM-21512 was implemented (4.7.30) but the link was not being displayed next to the recurring contribution because of this bit of code.

What it actually means is:
- BEFORE: Core did not allow editing a recurring contribution if linked to a membership.
- AFTER: Core does not restrict editing a recurring contribution if linked to a membership.

The "auto_renew" really just means the membership has an associated recurring contribution.